elastic-search-support/src/main/java/uk/gov/hmcts/ccd/definition/store/elastic/mapping/support/injection/TypeMappersManager.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.ccd.definition.store.elastic.mapping.support.injection;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.elastic.mapping.type.TypeMappingGenerator;
+
+import javax.annotation.PostConstruct;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.collect.Maps.newHashMap;
+
+/**
+ * used to inject type mappers, which can't be Autowired because of a circular dependency between them. i.e. a
+ * ComplexTypeMappingGenerator needs to be injected with a ComplexTypeMappingGenerator to handle complex types made
+ * of nested complex types
+ */
+@Component
+public class TypeMappersManager {
+
+    protected Map<String, TypeMappingGenerator> typeMappers = newHashMap();
+
+    //This line will guarantee the TypeMappersManager will be injected last
+    @Autowired
+    private Set<Injectable> injectables = new HashSet();
```

Comment:
assigning new HashSet(), is this intentional?

Reply:
👍